map/bookmark_manager.cpp

```diff
@@ -2207,7 +2207,25 @@ void BookmarkManager::UpdateBmGroupIdList()
   CHECK_THREAD_CHECKER(m_threadChecker, ());
   size_t const count = m_categories.size();
 
-  // All this routine to get sorted list by last modified time.
+  using PairT = std::pair<kml::MarkGroupId, BookmarkCategory const *>;
+  std::vector<PairT> vec;
+  vec.reserve(count);
+  for (auto const & [markGroupId, categoryPtr] : m_categories)
+    vec.emplace_back(markGroupId, categoryPtr.get());
+
+  m_bmGroupsIdList.clear();
+  m_bmGroupsIdList.resize(count);
+  for (size_t i = 0; i < count; ++i)
+    m_bmGroupsIdList[i] = vec[i].first;
```

Comment:
Will m_bmGroupsIdList always contain numbers from 1 to N, where N is a total categories count?

Reply:
Yes it will. This property is updated on the `addCategort` and `deleteCategory` actions (in the current implementation).
Or maybe I don't understand your question...